AISKU/Tests/Perf/src/AISKUPerf.Tests.ts

```diff
@@ -0,0 +1,362 @@
+    let metricVal = isProcessTime? "ProcessTime":"UsedJSHeapSize";
+    let unit = isProcessTime?  "ms":"KB";
+    let event =  {
+        name: "SDKPerfTest",
+        iKey: TENANT_KEY,
+        ver: "4.0",
+        ext: {},
+        baseData: {
+          testName: name,
+          sku:AISKUInitPerf.skuName,
+          version: AISKUInitPerf.version,
+          unitOfMeasure: unit,
+          metric: metricVal,
+          value: value
+        }
+    };
+
+    pushPerfEvents(event, AISKUInitPerf);
+    let message = msg? msg :`perfEvent: ${event.baseData.testName} ${event.baseData.value}${event.baseData.unitOfMeasure} added`;
+    console.log(message);
+    Assert.ok(true, message);
+}
+
+function parseBatchSendEvent(perfEvent: any, AISKUInitPerf: any, memoryUsageMarks: any): void {
+    let curMemory = performance["memory"]?.usedJSHeapSize;
+    let memoryMarks = Object.keys(memoryUsageMarks);
+    var index = "";
+    
+    if (memoryMarks && memoryMarks.length && curMemory !== undefined) {
+        index = memoryMarks[memoryMarks.length-1];
+        createPerfEvent(AISKUInitPerf, perfEvent.name+index, perfEvent?.exTime, true);
+    }
+}
+
+function parseTrackEvent(perfEvent: any, AISKUInitPerf: any): void {
+    let payloadItem = perfEvent["payload"]?.item;
+    let baseType = null;
+    if (perfEvent["time"] && payloadItem) {
+         baseType = payloadItem["baseType"];
+    }
+    // skip PageviewPerformanceData && MessageData
+    if (isNullOrUndefined(baseType) || baseType == "PageviewPerformanceData" || baseType == "MessageData") return;
+
+    createPerfEvent(AISKUInitPerf, `Track${baseType}`, perfEvent?.time, true);
+    var childEvts = perfEvent?.childEvts;
+    while (childEvts?.length) {
+        let curEvt = childEvts[0];
+        let name = `${baseType}-${curEvt.name}`;
+        let value = curEvt?.exTime !== undefined ? curEvt.exTime:curEvt.time;
+        createPerfEvent(AISKUInitPerf, name, value, true);
+        childEvts = curEvt?.childEvts;
+    }
+}
+
+function parseAppInsightsPerfEvent(perfEvent: any, AISKUInitPerf: any, memoryUsageMarks: any): void {
+    let perfEventName = perfEvent?.name;
+    switch (perfEventName) {
+        case "AISKU.flush":
+            parseBatchSendEvent(perfEvent, AISKUInitPerf, memoryUsageMarks);
+            break;
+        case "AppInsightsCore:track":
+            parseTrackEvent(perfEvent, AISKUInitPerf);
+            break;
+    }
+}
+
+function addSnippetLoadingTimeEvent(AISKUPerfTest: any, endtime: number): void {
+    let rawTime = endtime - AISKUPerfTest.snippetStartTime;
+    var duration = Math.round(rawTime*1000)/1000;
+    createPerfEvent(AISKUPerfTest, "SDKInit", duration, true, `AppInsightsInit-Init: Script LoadingTime: ${duration} ms added`);
+}
+
+
+export class AISKUPerf extends AITestClass {
+    public AISKUPerfTest: AppInsightsInitPerfTestClass;
+    protected TENANT_KEY: string = "2252db2e5e344635a36c5f1c04b3902c-eaced1c8-a046-4e8d-8fa8-c1ecf2077a5d-7256";
```

Comment:
This is already defined on line 51 -- we should just define once and reuse that one